Source/Widgets/Viewer/Viewer.js

```diff
@@ -2074,8 +2073,12 @@ function zoomToOrFly(that, zoomTarget, options, isFlight) {
   //We can't actually perform the zoom until all visualization is ready and
   //bounding spheres have been computed.  Therefore we create and return
   //a deferred which will be resolved as part of the post-render step in the
-  //frame that actually performs the zoom
-  const zoomPromise = defer();
+  //frame that actually performs the zoom.
+  const zoomPromise = new Promise((resolve) => {
```

Comment:
There is [an issue related to error handling for zoomTo](https://github.com/CesiumGS/cesium/issues/10222) that we're tracking. If it's a quick fix here, it would be great to fix it. But let me know if that's actually out-of-scope for this PR.

Reply:
Fixed it by catching the error in the `tileset.readyPromise` and calling `cancelZoom` in the `catch` block. Added a spec to make sure that `zoomTo` returns in the case of a failed tileset load - previously, it would never return.